src/parser/deserializer.ts

```diff
@@ -120,6 +123,47 @@ function deserializeObject(
   const promoteLongs = options['promoteLongs'] == null ? true : options['promoteLongs'];
   const promoteValues = options['promoteValues'] == null ? true : options['promoteValues'];
 
+  // Ensures default validation option if none given
+  const validation = options['validation'] == null ? { utf8: true } : options['validation'];
+
+  // Shows if global utf8 validation is enabled or disabled
+  let globalUTFValidation = true;
+  // Reflects utf8 validation boolean regardless of global or specific key validation
+  let uniformBool: boolean;
+  // Set of keys either to enable or disable validation on
+  const utf8KeysSet = new Set();
+
+  // Check for boolean uniformity and empty validation option
+  const keys = validation.utf8;
```

Comment:
Lower in this function this variable could be confused with the actual document keys so lets try: `utf8ValidatedKeys`